tests/neuro_san/internals/graph/activations/test_argument_assigner.py

```diff
@@ -0,0 +1,174 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from unittest import TestCase
+
+from neuro_san.internals.graph.activations.argument_assigner import ArgumentAssigner
+
+
+class TestArgumentAssigner(TestCase):
+    """Unit tests for ArgumentAssigner class"""
```

Comment:
Nice. Sorry to not have this kind of test in place earlier, it would have come in handy at a time like this.
That is an artifact of how this code came to be in the first place - by refactoring an experiment from Babak.
Theoretically almost all of the examples here would pass given the old code (and maybe a little more properties config).